File: 프로그래머스/unrated/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    map = {}
    answer = []
    for i in range(len(s)):
        x = s[i]
        if x in map:
            answer.append(i - map[x])
            map[x] = i
        else:
            map[x] = i
            answer.append(-1)
            
    return answer

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
function solution(s) {
    const map = new Map()
    var answer = [...s].map((el, i) => {
        if (map.has(el)) {
            const distance = i - map.get(el)
            map.set(el, i)
            return distance
        } else {
            map.set(el, i)
            return -1
        }
    })
    
    return answer;
}